weave-js/src/components/Form/SelectMultiple.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * Customized version of react-select for selecting multiple options.
+ */
+
+import React, {MouseEventHandler} from 'react';
+import {
+  components,
+  MultiValueGenericProps,
+  MultiValueProps,
+  Props as ReactSelectProps,
+} from 'react-select';
+import {
+  SortableContainer,
+  SortableContainerProps,
+  SortableElement,
+  SortableHandle,
+  SortEndHandler,
+} from 'react-sortable-hoc';
+
+import {AdditionalProps, Select} from './Select';
+
+function arrayMove<T>(array: readonly T[], from: number, to: number) {
+  const slicedArray = array.slice();
+  slicedArray.splice(
+    to < 0 ? array.length + to : to,
+    0,
+    slicedArray.splice(from, 1)[0]
+  );
+  return slicedArray;
+}
+
+const SortableMultiValueLabel = SortableHandle(
+  (props: MultiValueGenericProps) => <components.MultiValueLabel {...props} />
+);
+
+type Props<Option> = ReactSelectProps<Option, true> & AdditionalProps;
+
+export const SelectMultipleSortable = <Option,>(props: Props<Option>) => {
+  const SortableMultiValue = SortableElement(
+    (smvprops: MultiValueProps<Option>) => {
+      // this prevents the menu from being opened/closed when the user clicks
+      // on a value to begin dragging it. ideally, detecting a click (instead of
+      // a drag) would still focus the control and toggle the menu, but that
+      // requires some magic with refs that are out of scope for this example
+      const onMouseDown: MouseEventHandler<HTMLDivElement> = e => {
+        e.preventDefault();
+        e.stopPropagation();
+      };
+      const innerProps = {...smvprops.innerProps, onMouseDown};
+      return (
+        <components.MultiValue
+          {...smvprops}
+          innerProps={innerProps}
+          isDisabled={true}
+        />
+      );
+    }
+  );
+
+  const SortableSelect = SortableContainer(Select) as React.ComponentClass<
```

Comment:
maybe pop this outside as well, tho I see why you wanted to keep them in here to ensure these are only used in this component

Reply:
It was in order to use types the containing class is parameterized with. But I agree it's not clear - I just moved them out at the cost of some `any`s and I do like that better.